### PR TITLE
refactor: Platform Parsing and Game Collections

### PR DESCRIPTION
**Pull request by colin969**

### CODE REVIEW

File: src/shared/platform/interfaces.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  ReleaseDate?: string;         // (Nullable<DateTime>)
+  Version?: string;             // (String)
+  OriginalDescription?: string; // (String)
+  Language?: string;            // (String)
+}
+
+/**
+ * This interface represents an Additional Application as stored in Platform XMLs
+ * Comments represent their XML data types and IAdditionalApplicationInfo counterparts
+ * (NOTE: Best not to mess with the casing of the fields)
```
Comment (TBubba) [inline src/shared/platform/interfaces.ts]: I think the links to the Launchbox docs should be kept, at least until we stop using their XML format

File: src/shared/platform/interfaces.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ */
+export type IRawAdditionalApplicationInfo = {
+  Id: string;               // (String)
```
Comment (TBubba) [inline src/shared/platform/interfaces.ts]: You missed a space

File: src/shared/game/interfaces.ts [hunk 3]
```diff
@@ -75,8 +75,8 @@ export interface IAdditionalApplicationInfo {
    * (If false, this will only run when specifically launched)
    */
   autoRunBefore: boolean;
-  /** Command line argument(s) used when launching the additional application */
-  commandLine: string;
+  /** Command line argument(s) passed to the application to launch the game */
+  launchCommand: string;
```
Comment (TBubba) [inline src/shared/game/interfaces.ts]: I think these properties should have similar names to their XML counterparts

Comment (colin969) [reply src/shared/game/interfaces.ts]: We already refer to it as a Launch Command everywhere seen in the launcher ui, just not the code base, and we already refer to Game CommandLine as launchCommand in the code base too. I'd disagree.

Comment (TBubba) [reply src/shared/game/interfaces.ts]: I found 3 places where it was referred to as "launch command", and only one of them was visible to the end user. But I don't mind it too much so keep it like this if you want. I also noticed that there are some leftover language strings ("commandLine" and "noCommandLine")

File: src/renderer/components/RightBrowseSidebarAddApp.tsx [hunk 4]
```diff
@@ -26,11 +26,11 @@ export interface RightBrowseSidebarAddApp {
 
 /** Displays an additional application for a game in the right sidebar of BrowsePage. */
 export class RightBrowseSidebarAddApp extends React.Component<RightBrowseSidebarAddAppProps> {
-  onNameEditDone            = this.wrapOnTextChange((addApp, text) => { addApp.name = text; });
-  onApplicationPathEditDone = this.wrapOnTextChange((addApp, text) => { addApp.applicationPath = text; });
-  onCommandLineEditDone     = this.wrapOnTextChange((addApp, text) => { addApp.commandLine = text; });
-  onAutoRunBeforeChange     = this.wrapOnCheckBoxChange((addApp) => { addApp.autoRunBefore = !addApp.autoRunBefore; });
-  onWaitForExitChange       = this.wrapOnCheckBoxChange((addApp) => { addApp.waitForExit = !addApp.waitForExit; });
+  onNameEditDone              = this.wrapOnTextChange((addApp, text) => { addApp.name = text; });
```
Comment (TBubba) [inline src/renderer/components/RightBrowseSidebarAddApp.tsx]: These are indented 2 spaces more than they should

**Pull request merged**